apps/aehttp/test/aehttp_contracts_SUITE.erl

```diff
@@ -0,0 +1,1234 @@
+    aecore_suite_utils:connect(NodeName),
+
+    {ok, _} = aecore_suite_utils:mine_blocks(NodeName, 2),
+
+    %% Prepare accounts, Alice, Bert, Carl and Diana.
+
+    StartAmt = 5000,
+    {APubkey, APrivkey} = new_account(StartAmt),
+    {BPubkey, BPrivkey} = new_account(StartAmt),
+    {CPubkey, CPrivkey} = new_account(StartAmt),
+    {DPubkey, DPrivkey} = new_account(StartAmt),
+
+    {ok, [Block|_]} = aecore_suite_utils:mine_blocks(NodeName, 2),
+    ct:pal("Block ~p\n", [aec_blocks:txs(Block)]),
+
+    %% Save account information.
+    Accounts = #{acc_a => #{pub_key => APubkey,
+			    priv_key => APrivkey,
+			    start_amt => StartAmt},
+		 acc_b => #{pub_key => BPubkey,
+			    priv_key => BPrivkey,
+			    start_amt => StartAmt},
+		 acc_c => #{pub_key => CPubkey,
+			    priv_key => CPrivkey,
+			    start_amt => StartAmt},
+		 acc_d => #{pub_key => DPubkey,
+			    priv_key => DPrivkey,
+			    start_amt => StartAmt}},
+    [{accounts,Accounts},{node_name,NodeName}|Config];
+init_per_group(_Group, Config) ->
+    NodeName = aecore_suite_utils:node_name(?NODE),
+    aecore_suite_utils:start_node(?NODE, Config),
+    aecore_suite_utils:connect(NodeName),
+    ToMine = aecore_suite_utils:latest_fork_height(),
+    ct:pal("ToMine ~p\n", [ToMine]),
+    aecore_suite_utils:mine_blocks(NodeName, ToMine),
+    [{node_name,NodeName}|Config].
+
+end_per_group(all_endpoints, _Config) ->
+    ok;
+end_per_group(_Group, Config) ->
+    RpcFun = fun(M, F, A) -> rpc(?NODE, M, F, A) end,
+    {ok, DbCfg} = aecore_suite_utils:get_node_db_config(RpcFun),
+    aecore_suite_utils:stop_node(?NODE, Config),
+    aecore_suite_utils:delete_node_db_if_persisted(DbCfg),
+    ok.
+
+init_per_testcase(_Case, Config) ->
+    [{tc_start, os:timestamp()}|Config].
+
+end_per_testcase(_Case, Config) ->
+    Ts0 = ?config(tc_start, Config),
+    ct:log("Events during TC: ~p", [[{N, aecore_suite_utils:all_events_since(N, Ts0)}
+                                     || {_,N} <- ?config(nodes, Config)]]),
+    ok.
+
+%% ============================================================
+%% Test cases
+%% ============================================================
+
+%% null(Config)
+%%  Does nothing and always succeeds.
+
+null(_Config) ->
+    ok.
+
+%% spending_1(Config)
+%%  A simple test of tokens from acc_a to acc_b.
+
+spending_1(Config) ->
+    NodeName = proplists:get_value(node_name, Config),
+    %% Get account information.
+    #{acc_a := #{pub_key := APubkey,
+		 priv_key := APrivkey},
+      acc_b := #{pub_key := BPubkey}} = proplists:get_value(accounts, Config),
+
+    ct:pal("Top 1 ~p\n", [get_top()]),
+
+    %% Check initial balances.
+    ABal0 = get_balance(APubkey),
+    BBal0 = get_balance(BPubkey),
+    ct:pal("Balances 0: ~p, ~p\n", [ABal0,BBal0]),
+
+    %% Add tokens to both accounts.
+    {ok,200,_} = post_spend_tx(APubkey, 500, 1),
+    {ok,200,_} = post_spend_tx(BPubkey, 500, 1),
+    aecore_suite_utils:mine_blocks(NodeName, 2),
```

Comment:
It seems that this does not guarantee that the two spend tx at the lines just above are in the chain. (This is neither good nor bad on its own - it depends what the rest of the test case assumes.)

Reply:
This has been fixed with a mine_until.